### PR TITLE
Name filter

### PR DESCRIPTION
**Pull request by ChitvanRamani22**
## Related Issue 

- Info about the related issue 

Closes: #[issue number that will be closed through this PR]

### Describe the changes you've made

Give a clear description what modifications you have made

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Enhancement in the existing features
- [ ] Other

## Mention any unusual behaviour of your code (Write NA if not)
Any unusual behaviour of your code

## Checklist:

- [ ] My code follows the guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly whereever it was hard to understand.
- [ ] My changes generate no new warnings.
- [ ] I have verified/tested my code by running it locally.

## Additional Info (optional)
Any additional information you want to give

### CODE REVIEW

**Review by Ashish2030 — APPROVED** [review]

**Comment by ChitvanRamani22** [comment]
@Namanl2001 please have a look at this pr. It solves the issue https://github.com/Namanl2001/MERN-Gurujii-dev/issues/10

File: src/store/reducer/user.js [hunk 1]
```diff
@@ -66,6 +66,18 @@ function rootReducer(state = defaultState, action) {
         subject: 'All',
         class: 'All',
       };
+    case 'USERS':
+      const newArray8 = state.all.filter(
+        user => user.username == action.username
```
Comment (Namanl2001) [inline src/store/reducer/user.js:71]: You are defining newArray8 so better to shift it below newArray7.
Also make the filter case in-sensitive
use `===`

**Review by Namanl2001 — CHANGES_REQUESTED** [review]
Small nits else LGTM! Also, add some screenshots from the laptop and mobile screen as well. thnaks

**Comment by ChitvanRamani22** [comment]
The screenshots for laptop-view and mobile-view are attached below:

laptop-view
mobile-view

**Comment by Namanl2001** [comment]
In mobile, it is looking cool but on pc, I guess moving filters side by side would be nice. WDYT?

**Comment by ChitvanRamani22** [comment]
@Namanl2001 does this looks fine?

Screenshot (475)
Screenshot (476)

**Comment by Namanl2001** [comment]
> @Namanl2001 does this looks fine?

Yep! fine

File: src/components/filter.jsx [hunk 2]
```diff
@@ -1,58 +1,66 @@
...
+  handlePinChange = e => {
+    this.setState({
+      pin: e.target.value,
+    });
+  };
+
+  handleSubmitName = e => {
     e.preventDefault();
-    this.props.pin(this.state.content);
-    this.setState({ content: '' });
```
Comment (Namanl2001) [inline src/components/filter.jsx:19]: This should be present in `handleSubmitName( )` and `handleSubmitPIN( )`

Comment (sagar-barapatre) [reply src/components/filter.jsx:19]: 'content' state property is being replaced by 'name'. Hence she has already added this in handleSubmitPIN( ) as  this.props.pin(this.state.name);

**Comment by Namanl2001** [comment]
@ChitvanRamani22 why have made a new pr for these changes? What about this pr then?

**Comment by ChitvanRamani22** [comment]
@Namanl2001 i am.closing this.

**Pull request closed without merging**